lib/ansible/modules/cloud/amazon/cloudwatchlogs_log_group.py

```diff
@@ -0,0 +1,298 @@
+      default: present
+      required: false
+    log_group_name:
+      description:
+        - The name of the log group.
+      required: true
+    kms_key_id:
+      description:
+        - The Amazon Resource Name (ARN) of the CMK to use when encrypting log data.
+      required: false
+    tags:
+      description:
+        - The key-value pairs to use for the tags.
+      required: false
+    retention:
+      description:
+        - "The number of days to retain the log events in the specified log group.
+           Valid values are: [1, 3, 5, 7, 14, 30, 60, 90, 120, 150, 180, 365, 400, 545, 731, 1827, 3653]"
+      required: false
+    overwrite:
+     description:
+        - Whether an existing log group should be overwritten on create.
+     default: false
+     required: false
+extends_documentation_fragment:
+    - aws
+    - ec2
+'''
+
+EXAMPLES = '''
+# Note: These examples do not set authentication details, see the AWS Guide for details.
+
+- cloudwatchlogs_log_group:
+    log_group_name: test-log-group
+
+- cloudwatchlogs_log_group:
+    state: present
+    log_group_name: test-log-group
+    tags: { "Name": "test-log-group", "Env" : "QA" }
+
+- cloudwatchlogs_log_group:
+    state: present
+    log_group_name: test-log-group
+    tags: { "Name": "test-log-group", "Env" : "QA" }
+    kms_key_id: arn:aws:kms:region:account-id:key/key-id
+
+- cloudwatchlogs_log_group:
+    state: absent
+    log_group_name: test-log-group
+
+'''
+
+RETURN = '''
+log_groups:
+    description: Return the list of complex objetcs representing log groups
+    returned: success
+    type: complex
+    contains:
+        log_group_name:
+            description: The name of the log group.
+            returned: always
+            type: string
+        creation_time:
+            description: The creation time of the log group.
+            returned: always
+            type: integer
+        retention_in_days:
+            description: The number of days to retain the log events in the specified log group.
+            returned: always
+            type: integer
+        metric_filter_count:
+            description: The number of metric filters.
+            returned: always
+            type: integer
+        arn:
+            description: The Amazon Resource Name (ARN) of the log group.
+            returned: always
+            type: string
+        stored_bytes:
+            description: The number of bytes stored.
+            returned: always
+            type: string
+        kms_key_id:
+            description: The Amazon Resource Name (ARN) of the CMK to use when encrypting log data.
+            returned: always
+            type: string
+'''
+
+try:
+    import boto3
+    HAS_BOTO3 = True
+except ImportError:
+    HAS_BOTO3 = False
+
+import traceback
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.ec2 import _camel_to_snake, camel_dict_to_snake_dict, boto3_conn, ec2_argument_spec, get_aws_connection_info
```

Comment:
I'm not sure _camel_to_snake is stable to import and use. Are there cases when camel_dict_to_snake_dict isn't working right?